packages/expo-cli/src/commands/info/config/configAsync.ts

```diff
@@ -49,7 +50,13 @@ export async function actionAsync(projectRoot: string, options: Options) {
     });
   }
 
-  Log.log();
-  logConfig(options.full ? config : config.exp);
-  Log.log();
+  const configOutput = options.full ? config : config.exp;
+
+  if (!options.json) {
+    Log.log();
+    logConfig(configOutput);
+    Log.log();
+  } else {
+    Log.nested(JSON.stringify(configOutput));
```

Comment:
```suggestion
    Log.nested(JSON.stringify(configOutput, null, 2));
```

Reply:
I was thinking about maybe formatting the output the way you suggested, but that's what we already get without passing in `--json`. Compact json format made more sense to me.